src/components/Header/Header.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {
+  FunctionComponent,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+} from 'react';
+import classnames from 'classnames';
+import { Errors } from '../../types/Errors';
+import { addTodo } from '../../api/todos';
+import { User } from '../../types/User';
+import { AuthContext } from '../Auth/AuthContext';
+import {
+  useActiveTodos,
+  useCompleteTodos,
+  useOnFilteredStatusHandler,
+} from '../../hooks/hooks';
+import { TodosContext, TodosContextType } from '../Context/TodosContext';
+
+interface HeaderProps {
+  title: string;
+  setTitle: React.Dispatch<React.SetStateAction<string>>;
+  isAdding: boolean;
+  setIsAdding: React.Dispatch<React.SetStateAction<boolean>>;
+  setErrorMessage: React.Dispatch<React.SetStateAction<Errors>>;
+  loadingTodosFromServer: () => Promise<void>;
+  setSelectedTodosId: React.Dispatch<React.SetStateAction<number[]>>;
+}
+export const Header: FunctionComponent<HeaderProps> = ({
+  title,
+  setTitle,
+  isAdding,
+  setIsAdding,
+  setErrorMessage,
+  loadingTodosFromServer,
+  setSelectedTodosId,
+}) => {
+  const user = useContext<User | null>(AuthContext);
+  const { todos, setTodos } = useContext(TodosContext) as TodosContextType;
+
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const activeTodos = useActiveTodos(todos);
+  const completedTodos = useCompleteTodos(todos);
+  const onFilterStatusHandler = useOnFilteredStatusHandler(
+    setTodos, setErrorMessage, setSelectedTodosId,
+  );
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, []);
+
+  const onChangeHandler = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => setTitle(e.target.value),
+    [],
+  );
+
+  const onSubmitHandler = useCallback(
```

Comment:
```suggestion
  const handleChange= useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => setTitle(e.target.value),
    [],
  );

  const handleSubmit = useCallback(
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364